sympy/integrals/integrals.py

```diff
@@ -1308,10 +1308,13 @@ def principal_value(self):
         if a == b:
             return 0
         r = Dummy('r')
-        n = Dummy('n')
         u = Dummy('u')
 
-        singularities_list = [s for s in singularities(self.function, x) if s.is_comparable and a < s < b]
+        singularities_list = [s for s in singularities(self.function, x) if s.is_comparable and a <= s <= b]
+
+        for i in singularities_list:
+            if ((i is b) or (i is a)):
```

Comment:
Are these tested? It is generally better to use `i == b or i == a`. The text could say that the principal value is not defined because of singularity at `i`. That should probably be on two lines.

Reply:
@jksuom Yes, it is tested in line 35 in `test_integrals.py`. Okay, I'll add what you recommended..